xds/internal/xdsclient/client_new.go

```diff
@@ -100,6 +102,22 @@ func NewWithConfigForTesting(config *bootstrap.Config, watchExpiryTimeout, autho
 	return cl, grpcsync.OnceFunc(cl.close), nil
 }
 
+func init() {
+	internal.TriggerXDSResourceNameNotFoundClient = triggerXDSResourceNameNotFoundClient
+}
+
+var (
+	singletonClientForTestingMu sync.Mutex
+	singletonClientForTesting   *clientRefCounted
```

Comment:
It seems like this could be an atomic instead?

Reply:
Done.